### DISCUSSION TITLE
howto disable unsupported critical X.509 extension (critical 'msApplicationCertPolicies' extension not supported) on the server ?

### DISCUSSION THREAD

**Discussion by landjalan** [strongswan/strongswan#2516] [General]
hi, we have been working with strongswan/ikev2 server for a while (windows as server main os + linux clients)

to circumvent x509 extension issues we created the file /etc/strongswan.d/libstrongswan-relax-constraints.conf with the following content on the clients and could happily connect:
`libstrongswan {
    x509 {
         enforce_critical = no
   }
}`

now we switched  to a pfsense firewall with integrated strongswan (freebsd 14) which leads to getting the afore mentioned error in the logfiles as well but now server sided:

`
OpenSSL X.509 parsing failed,
found unsupported critical X.509 extension: 1.3.6.1.4.1.311.21.10,
critical 'msApplicationCertPolicies' extension not supported
`

but creating /var/etc/ipsec/strongswan.d/libstrongswan-relax-constraints.conf does obviously not help on freebsd

any ideas how to disable enforce_critical on a strongswan linux service ?

Thanks a lot

**Comment by tobiasbrunner** [comment]
> but creating /var/etc/ipsec/strongswan.d/libstrongswan-relax-constraints.conf does obviously not help on freebsd

This only depends on whether that file is included in the main `strongswan.conf` file. So wherever that's located on FreeBSD, check what `include` instructions it already contains (or just change the configuration directly in strongswan.conf, although that might get overwritten by pfsense, so check if they have any particular includes for custom overrides or some other mechanism to add custom configuration).

**Comment by landjalan** [comment]
thanks for your input !
on the working client, as well on the not working server "swanctl --stats" shows that **x509** and **constraints** are being loaded. but i could not figure out yet, how to check what the currently set parameters of the modules are (whether **enforce_critical = no** is active/enabled for the x509 module)

what also strikes me as odd is the fact that on my working client (where enforce_critical = no is working fine) the strongswan.conf is very small and under plugins does only include strongswan.d/charon/*.conf. but the actual file libstrongswan-relax-constraints.conf does lie in /strongswan.d/charon ...

**Reply by tobiasbrunner** [reply]
> but i could not figure out yet, how to check what the currently set parameters of the modules are (whether **enforce_critical = no** is active/enabled for the x509 module)

There is no status output or log message for this (unless the option is enabled and an unknown critical extension is found, which triggers an error).


> what also strikes me as odd is the fact that on my working client (where enforce_critical = no is working fine) the strongswan.conf is very small and under plugins does only include strongswan.d/charon/*.conf. but the actual file libstrongswan-relax-constraints.conf does lie in /strongswan.d/charon ...

If `strongswan.conf` is in `/`, there is nothing odd about that (`includes` are resolved relative to the file, see [the docs](https://docs.strongswan.org/docs/5.9/config/strongswanConf.html#_including_files)).

**Comment by landjalan** [comment]
once we got the radius server configuration right those x509 error messages in the serverlog disappeared, the x509 certificates are being parsed through to the windows ad radius server as expected. problem solved, issue gone.
